api/types/registry/registry.go

```diff
@@ -102,3 +105,14 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// ManifestInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+// TODO(nishanttotla): Decide which other fields to add here
+type ManifestInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
```

Comment:
@aaronlehmann as an example, what do you think about
```
{
  Digest       digest.Digest,
  Manifest     distribution.Manifest,
  ConfigJSON   []bytes,
}
```
I still think we'd have to separate cases (fat, v2, v1 manifests) to get the right config blob, but then the client could Unmarshal it as desired.